File: XTrailsJPA/src/main/java/com/mattaldrete/enitities/Location.java
package com.mattaldrete.enitities;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

public class Location {
	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int id;
	
	private String city;
	private String park;
	private String state;
	private double longitude;
	private double latitude;
	private String county;
	private String street;
	private int zipcode;
	
	
	

}

File: ngXTrails/src/app/entities/location/location.ts
import { User } from '../user/user';
import { zip } from 'rxjs';

export class Location {
  id: number;
  city: string;
  park: string;
  state: string;
  longitude: number;
  latitude: number;
  county: string;
  street: string;
  zipcode: number;
  trail: trail;
  user: User;


  constructor(
    id?: number,
    city?: string,
    state?: string,
    longitude?: number,
    latitude?: number,
    county?: string,
    street?: string,
    zipcode?: number,
    trail?: trail,
    user?: User
  ) {
    this.id = id;
    this.city = city;
    this.state = state;
    this.longitude = longitude;
    this.latitude = latitude;
    this.county = county;
    this.street = street;
    this.zipcode = zipcode;
    this.trail = trail;
    this.user = user;

  }
}
